internal/eventtracker/eventtracker.go

```diff
@@ -0,0 +1,146 @@
+type TrackerClient interface {
+	TrackEvent(ctx context.Context, eventName string, tags Tags) error
+}
+
+// CommandTracker is a basic implementation of the Tracker interface.
+type CommandTracker struct {
+	client TrackerClient
+}
+
+// NewCommandTracker creates a new CommandTracker instance.
+func NewCommandTracker(client TrackerClient) *CommandTracker {
+	return &CommandTracker{
+		client: client,
+	}
+}
+
+// NewPosthogTrackerClient creates a new PosthogTracker instance.
+func NewPosthogTrackerClient(apiKey string, serverURL string) *PosthogTracker {
+	client, _ := posthog.NewWithConfig(apiKey, posthog.Config{
+		Endpoint: serverURL,
+	})
+
+	return &PosthogTracker{
+		client: client,
+	}
+}
+
+type NoopTracker struct{}
+
+func NewNoopTrackerClient() *NoopTracker {
+	return &NoopTracker{}
+}
+
+func (n *NoopTracker) TrackEvent(_ context.Context, _ string, _ Tags) error {
+	return nil
+}
+
+// PosthogTracker is an implementation of the TrackerClient interface for PostHog.
+type PosthogTracker struct {
+	client posthog.Client
+}
+
+// TrackEvent sends an event to the PostHog server.
+func (p *PosthogTracker) TrackEvent(_ context.Context, eventName string, tags Tags) error {
+	msg := posthog.Capture{
+		DistinctId: "default-id",
+		Event:      eventName,
+	}
+
+	defaultTags := loadDefaultTags()
+	for k, v := range defaultTags {
+		msg.Properties.Set(k, v)
+	}
+
+	for k, v := range tags {
+		msg.Properties.Set(k, v)
+	}
+
+	machineID, err := machineid.ProtectedID("chainloop")
+	if err == nil {
+		msg.Properties.Set("machine_id", machineID)
+		msg.DistinctId = machineID
+	}
+
+	return p.client.Enqueue(msg)
+}
+
+// loadDefaultTags returns a map of default tags that are added to every event.
+func loadDefaultTags() Tags {
+	return Tags{}.WithRuntimeInfo().WithEnvironmentInfo()
+}
+
+// WithRuntimeInfo adds runtime information to the Tags.
+func (tg Tags) WithRuntimeInfo() Tags {
+	tg["os"] = runtime.GOOS
+	tg["arch"] = runtime.GOARCH
+
+	return tg
+}
+
+// WithEnvironmentInfo adds environment information to the Tags.
+func (tg Tags) WithEnvironmentInfo() Tags {
+	isCIValue := "false"
+	if runningOnCI() {
+		isCIValue = "true"
+	}
+	tg["ci"] = isCIValue
+
+	// TODO: Add more environment information for each individual CI system
+	switch crafter.DiscoverRunner(zerolog.Nop()).ID() {
+	case schemaapi.CraftingSchema_Runner_GITHUB_ACTION:
+		tg["runner"] = "GitHub Actions"
+	case schemaapi.CraftingSchema_Runner_GITLAB_PIPELINE:
+		tg["runner"] = "GitLab CI"
+	case schemaapi.CraftingSchema_Runner_AZURE_PIPELINE:
+		tg["runner"] = "Azure Pipelines"
+	case schemaapi.CraftingSchema_Runner_JENKINS_JOB:
+		tg["runner"] = "Jenkins"
+	case schemaapi.CraftingSchema_Runner_CIRCLECI_BUILD:
+		tg["runner"] = "CircleCI"
+	case schemaapi.CraftingSchema_Runner_DAGGER_PIPELINE:
+		tg["runner"] = "Dagger"
+	default:
+		tg["runner"] = "Unknown"
+	}
+
+	return tg
+}
+
+// runningOnCI returns true if the code is running in a CI environment.
+func runningOnCI() bool {
```

Comment:
I think this method should be based on our runner discovery. basically that the result is not unknown or generic I think. Whatever DiscoverRunner returns when it does not find it.

Reply:
We might nto need this method anymore, just check the output of discover